int32 state
float32 param1
float32 param2
float32 param3
